src/shared/state/balances.ts

```diff
@@ -0,0 +1,17 @@
+import { ViewService } from '@penumbra-zone/protobuf';
+import { BalancesResponse } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { penumbra } from '@/shared/penumbra';
+import { connectionStore } from '@/shared/state/connection';
+import { useQuery } from '@tanstack/react-query';
+
+const fetchQuery = async (): Promise<BalancesResponse[]> => {
+  return Array.fromAsync(penumbra.service(ViewService).balances({}));
+};
+
+export const useBalances = () => {
```

Comment:
docs: we should had a code comment that specifies that consumers of this hook are required to be wrapped in an `observer()` func